### PR TITLE
Feature: Calendar Days

### PR DESCRIPTION
**Pull request by cihancimen**
Adds days property to retail calendar. It is an ordered list of all days in the calendar year.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -74,6 +74,20 @@ export interface RetailCalendar {
   numberOfWeeks: number
   months: RetailCalendarMonth[]
   weeks: RetailCalendarWeek[]
+  days: RetailCalendarDay[]
+}
+
+export interface RetailCalendarDay {
```
Comment (cihancimen) [inline src/types.ts:80]: @cubbuk Do you think the  fields on this interface sufficient?

File: src/types.ts [hunk 2]
```diff
@@ -74,6 +74,20 @@ export interface RetailCalendar {
...
+  weekOfYear: number;
+  monthOfYear: number;
+  dayOfWeek: number // 1 = Monday, 7 = Sunday
```
Comment (cubbuk) [inline src/types.ts:83]: will this be 1th index similar to month?

File: src/types.ts (hunk 1 above)
Comment (cubbuk) [reply src/types.ts:80]: I think that would be enough for us.

File: src/types.ts (hunk 2 above)
Comment (cihancimen) [reply src/types.ts]: I just updated the comment. Now 1 = first day of week, 7 = last day of week. So it doesn't correspond to Mon-Fri.

**Review by cubbuk — APPROVED** [review]
💯

**Comment by cihancimen** [comment]
> @cihancimen which field will be used to understand whether the calendar is on day level? are we going to use calendarType similar to how we use gregorian?

Yes, but that option won't be visible to this repo.

**Pull request merged**